### ISSUE TITLE
how to reproduce 93.3% accuracy for NMSC dataset using 'skt/kogpt2-base-v2'?

### ISSUE THREAD

**Issue by dsindex** [SKT-AI/KoGPT2#36]
i had some experiments on [NMSC dataset](https://github.com/e9t/nsmc) with '[skt/kogpt2-base-v2](https://huggingface.co/skt/kogpt2-base-v2)' model.

but just got 89.55% accuracy on it.

(in addition, i got same results with GPT2Model or GPT2LMHeadModel.
 i think this is very natural since we are not using the lm head but using the output(hidden_states[-1]) of transformers. i.e. 
 the last token embedding of the last layer.)

could you share hyper-parameters(e.g. batch size, learning rate, etc) or some specific methods you took ?

**Comment by haven-jeon** [comment]
In the case of NSMC performance, larger batches and more training iterations showed good performance. We are preparing to release the code, but please understand that it can be difficult to reproduce on a typical machine. We are organizing it, so we would appreciate it if you could wait for the code to be released.

**Comment by dsindex** [comment]
thank you for response!
i’ll close this issue and wait for it.

**Comment by haven-jeon** [comment]
https://github.com/haven-jeon/KoGPT2-subtasks

Thanks.

**Comment by dsindex** [comment]
- i found somewhat weird results when calculating accuracy using `metric_acc.compute() in validation_epoch_end()`.

https://github.com/dsindex/KoGPT2-subtasks/blob/main/model/kogpt2.py#L141
(forked version for modification)
```
def validation_epoch_end(self, outputs):
        avg_loss = torch.stack([x["loss"] for x in outputs]).mean()
        val_acc = self.metric_acc.compute()
        self.log('val_loss', avg_loss)
        self.log('val_acc', val_acc, on_epoch=True, prog_bar=True) <-- HERE
...
```

we can see the validation accuracy(val_acc) `0.925 ~ 0.932` as expected.
```
... loss=0.309, v_num=0, val_acc=0.925, val_acc_2=0.890, train_loss_step=0.180, train_acc=0.929, train_loss_epoch=0.163]...
```

however, evaluation accuracy with saved model was just around `0.889`.
```
$ ./test-nsmc.sh
...
precision recall f1-score support

           0 0.8842 0.8948 0.8895 24827
           1 0.8950 0.8845 0.8897 25173

    accuracy 0.8896 50000
   macro avg 0.8896 0.8896 0.8896 50000
weighted avg 0.8897 0.8896 0.8896 50000

[[22215 2612]
 [ 2908 22265]]
```

that's why i add 'val_acc_2' to compute accuracy using `sklearn`.
https://github.com/dsindex/KoGPT2-subtasks/blob/main/model/kogpt2.py#L151
```
  def validation_epoch_end(self, outputs):
        avg_loss = torch.stack([x["loss"] for x in outputs]).mean()
        val_acc = self.metric_acc.compute()
        self.log('val_loss', avg_loss)
        self.log('val_acc', val_acc, on_epoch=True, prog_bar=True)
        self.metric_acc.reset()

        y_true = []
        y_pred = []
        for i in outputs:
            y_true += i["y_true"]
            y_pred += i["y_pred"] 
        val_acc_2 = accuracy_score(y_true, y_pred)  <-- HERE
        self.log('val_acc_2', val_acc_2, on_epoch=True, prog_bar=True)

        print(classification_report(y_true, y_pred, digits=4))
        print(confusion_matrix(y_true, y_pred))
```

where does the difference come from?

- source version used
```
pytorch_lightning : 1.2.10, 1.3.0
pytorch : 1.7.1+cu101
transformers : 4.5.1
```
